src/traps.rs

```diff
@@ -0,0 +1,25 @@
+#[derive(PartialEq, Debug)]
+pub enum TrapCode {
+    GETC = 0x20,
+    OUT = 0x21,
+    PUTS = 0x22,
+    IN = 0x23,
+    PUTSP = 0x24,
+    HALT = 0x25,
+}
+
+impl TryFrom<u16> for TrapCode {
```

Comment:
Maybe we could change the differents `TryFrom`s to just a `From` so we don't return a `Result`, if we encounter a wrong code feels legit to just do a `panic!` at the `_ =>` match arm, and make the code a little clearer without the `unwraps()`

Reply:
Mmm I don't totally agree. It is true that, atm, the vm's implementation does not account for wrong opcodes and other anomalies, but I think it's better to have a `TryFrom` for two reasons:
1. It is more idiomatic, more "rusty" if you like. By using `TryFrom`, it's super extra clear that the conversion `u16` -> `TrapCode` / `Opcode` is not guaranteed to succeed, which forces the consumer to account for that fact (even if they will just `unwrap`, at least it's their responsability).
2. In the ISA, there are exceptions to deal with wrong opcodes, memory violations and stuff like that. For now, it's not implemented, but I think keeping the errors in the domain of the program allows to more easily deal with those exceptions later, if/when they are implemented.

Let me know if you are not convinced!

Reply:
I have to say that in my project I implemented `From` (with `panic!`s) instead of `TryFrom`, but I'm with @lazcanoluca on this one. Of course the `Result`s should be handled properly, not just `unwrap`ing them.